test/integration/singlecluster/controller/jobs/job/job_controller_test.go

```diff
@@ -1028,8 +1029,16 @@ var _ = ginkgo.Describe("When waitForPodsReady enabled", ginkgo.Ordered, ginkgo.
 					// also updated due to setting of the job status.
 					g.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).Should(gomega.Succeed())
 					g.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, nil)).Should(gomega.Succeed())
+					g.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).Should(gomega.Succeed())
+					g.Expect(createdWorkload.Status.Conditions).To(gomega.ContainElements(
+						gomega.BeComparableTo(metav1.Condition{
+							Type:    kueue.WorkloadAdmitted,
+							Status:  metav1.ConditionFalse,
+							Reason:  "NoReservation",
+							Message: "The workload has no reservation",
+						}, util.IgnoreConditionTimestampsAndObservedGeneration),
+					))
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
-				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
```

Comment:
@PBundyra why is this line removed?

Reply:
I added the check above which ensures the condition is synced

Reply:
Is it effectively doing the same?

Reply:
Maybe separate is better, not because the change is bad, but because the role of the change is non-obvious in the PR with so many other changes, so it may distract future readers.

Reply:
I added this because one of the integration test was flaky because the Workload sometimes was still admitted when generating a new PodsReady condition. So this change is directly related to the whole PR